CanvasCore/CanvasCore/WhizzyWig/CanvasWebView.js

```diff
@@ -29,3 +29,40 @@ var checkQuizzesNext = setInterval(function() {
      }
   })
 }, 100)
+
+// Replace all iframes with a button to launch in SFSafariViewController
+const iframes = document.querySelectorAll('iframe');
+// If there is only one iframe
+// and id="cnvs_content"
+// and the src is a canvas file
+// reload the webview with an authenticated version of the iframe's src
+if (iframes.length == 1 && /\/courses\/\d+\/files\/\d+\/download/.test(iframes[0].src) && iframes[0].id === "cnvs_content") {
+    window.webkit.messageHandlers.loadFrameSource.postMessage(iframes[0].src)
+} else {
+    // Replace all iframes with a button to launch in SFSafariViewController
+    iframes.forEach(iframe => {
+        const replace = iframe => {
+            const a = document.createElement('a')
+            a.textContent = '{$LTI_LAUNCH_TEXT$}'
```

Comment:
does this get replaced with the localized text?

Reply:
Good call. It's now getting translated but it's not getting styled. I need to make sure the CSS gets loaded outside of PageTemplateRenderer. I think I might want to wait for your changes [here](https://github.com/instructure/canvas-ios/pull/1361) first though. what do you think?